### ISSUE TITLE
backend missing? A question about native build and R

### ISSUE THREAD

**Issue by c2b2pss** [drpowell/degust#74]
Total noob...trying to build this app natively to try and modify R to my code. Being a noob I cannot find the R directory and my build does not do the next steps after data import. 

Any help about how to integrate R would be greatly appreciated.  I cannot figure this out!

Many many thanks in advance!

Rendered degust/_libraries.R.erb (1.0ms)
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc]   Rendered degust/_common.R.erb (438.0ms)
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc]   Rendered degust/voom.R.erb (873.0ms)
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc] Completed 500 Internal Server Error in 1414ms (ActiveRecord: 1.0ms)
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc]
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc] **Errno::ENOENT (No such file or directory - R):**
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc]
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc] app/models/degust_logic.rb:65:in `run_r_code'
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc] app/controllers/degust_controller.rb:95:in `block in dge'
[e9f00a07-5a3d-4fa7-8536-d05cd797a9bc] app/controllers/degust_controller.rb:91:in `dge'

**Comment by drpowell** [comment]
The degust back-end expects R to be available in your PATH.  The error you are hitting means it cannot execute R on the system.  You'll need to install R, and the necessary R libraries:
`Rscript -e "install.packages(c('BiocManager','jsonlite')); BiocManager::install(version='3.9', ask=F); BiocManager::install(c('limma','edgeR'))"`

Alternatively, you can build the docker container as described in `INSTALL.md` which will install all the necessary dependencies.

Cheers

**Comment by c2b2pss** [comment]
Dear Dr. Powell:

Thank you for your reply.

Yes, R is installed and I am using it by itself.


1.      However, I cannot find the place where I can edit the $PATH info in the Degust code

2.      I am trying to build and install locally.

3.      The Docker version works quite nicely, but the a local build needs to be told where R lives. I cannot find the code to edit the path.  The Rscript only installs R locally, correct?. Or I cannot see where it is called in the code??

Sincerely,

-Prem Subramaniam

**Comment by drpowell** [comment]
The rails server should pickup `$PATH` from the external environment.   How are you running rails - and is R in the path at that point?

The place R is run is here : https://github.com/drpowell/degust/blob/master/app/models/degust_logic.rb#L65

Cheers

**Comment by c2b2pss** [comment]
Dear Dr. Powell:

After a brief hiatus, I came back to building a local copy of Degust….and I succeeded finally.  It runs nicely.  Quite nice really! I have been a pure biologist my entire career, and only now exploring the world of R and gene expression to learn.  Degust is a big help to bring my own thinking to analysis of my data than leaving it to someone else.

I noticed one issue that can be changed on the GitHub instructions, and I have a couple questions:


  1.  The Gemfile refers to Ruby 2.4.6, but your github instructions still say 2.4.0.  I had to install v2.4.6 and then it was fine. On MacOS.

My questions:


  1.  I am a semi-beginner so the question may not be framed right. Is it possible to somehow grab an output of the Configuration file end results so it can be replicated in R?  The huge advantage of Degust is to be able read metadata and create groups  with a few clicks.  I presume this is being done in java/json – is this runnable under R using jsonlite?
  2.  Can the .erb files be modified to perform custom R functions?

Again, many thanks for your replies, and of course an excellent program.

-Prem
